Add networking workgroup charter (#1424)
apple/swift-org-website: networking-workgroup/index.md

---
layout: page
title: Networking Workgroup
---

The Networking Workgroup is a team that guides the evolution of networking libraries, protocols, and APIs in the Swift ecosystem.

The Networking Workgroup will:

* Govern the development of Swift's networking libraries, including the HTTP client and server APIs, core networking primitives, and shared protocol implementations.
* Conduct community reviews of feature and API proposals for projects under the workgroup's purview.

> * Pursue the long-term directions outlined in the Networking vision, including a unified layered networking architecture for Swift.

Reviewer: Link the vision doc?

Some areas of frequent collaboration with other groups include, but are not limited to:

* Maintainers of SwiftNIO and projects built on top of it, such as AsyncHTTPClient, gRPC-swift, Vapor, and Hummingbird
* Maintainers of platform-specific networking frameworks and HTTP stacks, such as Network.framework and URLSession
* The Swift Server Workgroup on HTTP server frameworks and the broader Swift Server ecosystem
* The [Build and Packaging Workgroup](/build-and-packaging-workgroup) on package-level networking concerns
* Cross-platform I/O and networking support with the [Platform Steering Group](/platform-steering-group)

### Scope

The workgroup's scope spans the networking stack: from I/O primitives and transport protocols such as QUIC and TLS, through protocol implementations, to high-level client and server APIs. The following are explicitly out of scope:

* Configuring system-wide proxy settings
* Application-layer frameworks built on top of HTTP (web frameworks, ORMs, templating engines) — though the workgroup aims to provide foundations those frameworks can build on

### Relationship with the Swift Server Workgroup

> The Swift Server Workgroup (SSWG) promotes the use of Swift for server application development and runs an incubation process for server-oriented libraries. There is natural overlap: many of the networking libraries the Networking Workgroup governs are heavily used in server contexts, and several SSWG-incubated packages (such as Async HTTP Client) are networking libraries.

Reviewer 2: Removes spacing in `AsyncHTTPClient` name to be consistent across the page.

```suggestion
The Swift Server Workgroup (SSWG) promotes the use of Swift for server application development and runs an incubation process for server-oriented libraries. There is natural overlap: many of the networking libraries the Networking Workgroup governs are heavily used in server contexts, and several SSWG-incubated packages (such as AsyncHTTPClient) are networking libraries.
```

* The Networking Workgroup provides the foundational networking libraries, APIs, and protocol implementations that the ecosystem builds on, and guides their evolution to serve the entire Swift ecosystem — apps, servers, embedded, CLI tools, and more.
* The Swift Server Workgroup is a key stakeholder and one of the primary consumers of these foundations, with a specific focus on the broader server application ecosystem: incubating libraries, promoting best practices, and channeling server-specific needs (database drivers, deployment tooling, service lifecycle, etc.).

In practice, the groups will collaborate closely:

* Networking proposals that significantly affect server use cases will be shared with the SSWG for input before or during review, and vice versa for proposals that significantly affect networking.
* Members may serve on both groups to ensure alignment, and joint meetings may be held when topics span both charters.

### Evolution

The Ecosystem Steering Group has delegated the purview over the workgroup's repositories to the Networking Workgroup. The Networking Workgroup uses the Swift Evolution process for its repositories.

> Proposals or vision documents that intersect with platform specific behavior will be reviewed in collaboration with the Platform Steering Group.

Reviewer: platform-specific

Reviewer 2: ```suggestion
Proposals or vision documents that intersect with platform-specific behavior will be reviewed in collaboration with the Platform Steering Group.
```

The Ecosystem Steering Group designates one member of the workgroup as the chair. The chair has no special authority over the workgroup, but they are responsible for ensuring its smooth functioning, including by:

* organizing and leading regular meetings
* ensuring that the workgroup communicates effectively with the community
* coordinating meetings between workgroup representatives and other Swift workgroups or teams when necessary

If you'd like to join the workgroup, send a message to [@networking-workgroup](https://forums.swift.org/new-message?groupname=networking-workgroup) on the Forums and you will be invited to the next available group meeting to discuss it. See [Community Participation](#community-participation) for examples of ways to contribute and demonstrate your interest to the group.

Workgroup members will try to make decisions independently by consensus whenever possible, and will raise issues to the Ecosystem Steering Group when there are particular challenges with reaching consensus on significant decisions.

## Meetings

> The Networking Workgroup meets biweekly. Meetings take place in odd/even numbered weeks, unless otherwise communicated.

Reviewer: You're going to have to pick one of "odd/even". 😉

> The Networking Workgroup meets biweekly. Meetings take place in odd/even numbered weeks, unless otherwise communicated.

Reviewer 2: Simplifying this line, changing biweekly -> "every two weeks" and removing the mention of odd/even numbered weeks which was unclear.

```suggestion
The Networking Workgroup meets every two weeks, unless otherwise communicated.
```
